### PR TITLE
Fix Inconsistencies with text when toggling lights while being  inside a closet for example.

### PR DESCRIPTION
**Pull request by Huffie56**
# About the pull request
Fix Inconsistencies with text when toggling lights while being inside a closet for example.

fixes https://github.com/cmss13-devs/cmss13/issues/1083

autocloses on merge

# Explain why it's good for the game
resolve bug https://github.com/cmss13-devs/cmss13/issues/1083

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

image

</details>

# Changelog

:cl:

fix: Fix Inconsistencies with text when toggling lights while being inside a closet for example.

/:cl:

### CODE REVIEW

**Comment by ItsVyzo** [comment]
@Huffie56 change your text to exactly the following:

fixes #1083 

autocloses on merge

File: code/game/objects/items/devices/flashlight.dm [hunk 1]
```diff
@@ -59,8 +59,12 @@
 		to_chat(user, SPAN_WARNING("You cannot toggle \the [src.name] on or off."))
 		return FALSE
 	if(!isturf(user.loc))
-		to_chat(user, "You cannot turn the light on while in [user.loc].") //To prevent some lighting anomalities.
-		return FALSE
+		if(on==TRUE)
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/objects/items/devices/flashlight.dm]: done.

File: code/game/objects/items/devices/flashlight.dm [hunk 2]
```diff
@@ -59,8 +59,12 @@
 		to_chat(user, SPAN_WARNING("You cannot toggle \the [src.name] on or off."))
 		return FALSE
 	if(!isturf(user.loc))
-		to_chat(user, "You cannot turn the light on while in [user.loc].") //To prevent some lighting anomalities.
-		return FALSE
+		if(on)
+			to_chat(user, SPAN_WARNING("You cannot turn the light off while in [user.loc].")) //To prevent some lighting anomalities.
```
Comment (BeagleGaming1) [inline code/game/objects/items/devices/flashlight.dm:63]: instead of the if and else, couldn't you just make it use the "on" variable? Something like `"You cannot turn the light [on ? "off" : "on"] while in [user.loc]."`.

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/flashlight.dm (hunk 2 above)
Comment (Huffie56) [reply code/game/objects/items/devices/flashlight.dm:63]: if(!isturf(user.loc))
		to_chat(user, SPAN_WARNING("You cannot turn the light off or on while in [user.loc].")) //To prevent some lighting anomalities.

i replace my code with this seem more simpler and work.

File: code/modules/clothing/head/misc_special.dm [hunk 3]
```diff
@@ -130,10 +130,10 @@
 	..()
 
 	if(!isturf(user.loc))
-		to_chat(user, "You cannot turn the light on while in [user.loc]") //To prevent some lighting anomalities.
-		return
+		to_chat(user, SPAN_WARNING("You cannot turn the light [on ? "off" : "on" ] while in [user.loc].")) //To prevent some lighting anomalies.
+		return FALSE
```
Comment (BeagleGaming1) [inline code/modules/clothing/head/misc_special.dm:134]: Why was this changed from return to return FALSE?

File: code/modules/clothing/head/misc_special.dm [hunk 4]
```diff
@@ -130,10 +130,10 @@
...
+
 	on = !on
-	icon_state = "hardhat[on]_pumpkin"
```
Comment (BeagleGaming1) [inline code/modules/clothing/head/misc_special.dm:136]: why was this icon_state change removed?

File: code/modules/clothing/suits/marine_armor.dm [hunk 5]
```diff
@@ -206,8 +206,9 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
 	..()
 
 	if(!isturf(user.loc))
-		to_chat(user, SPAN_WARNING("You cannot turn the light on while in [user.loc].")) //To prevent some lighting anomalities.
-		return
+		to_chat(user, SPAN_WARNING("You cannot turn the light [is_light_on() ? "off" : "on"] while in [user.loc].")) //To prevent some lighting anomalies.
+		return FALSE
```
Comment (BeagleGaming1) [inline code/modules/clothing/suits/marine_armor.dm:210]: Why was this changed from return to return FALSE?

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

File: code/modules/clothing/suits/marine_armor.dm (hunk 5 above)
Comment (Huffie56) [reply code/modules/clothing/suits/marine_armor.dm:210]: removed

File: code/modules/clothing/head/misc_special.dm (hunk 3 above)
Comment (Huffie56) [reply code/modules/clothing/head/misc_special.dm:134]: removed

File: code/modules/clothing/head/misc_special.dm (hunk 4 above)
Comment (Huffie56) [reply code/modules/clothing/head/misc_special.dm:136]: added back

**Review by BeagleGaming1 — APPROVED** [review]
code looks good
